gson/src/main/java/com/google/gson/internal/bind/JsonTreeWriter.java

```diff
@@ -199,6 +200,16 @@ private void put(JsonElement value) {
     return this;
   }
 
+  /**
+   * {@inheritDoc}
+   *
+   * @since 2.8.7
+   */
+  @Override public JsonWriter jsonValue(String value) throws IOException {
+    put(new JsonValue(value));
+    return this;
```

Comment:
Should check if `value == null` and then emit JSON null, like it is done by `JsonWriter.jsonValue(...)`.
```suggestion
    if (value == null) {
      return nullValue();
    } else {
      put(new JsonValue(value));
      return this;
    }
```